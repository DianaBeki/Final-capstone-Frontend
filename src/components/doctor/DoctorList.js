import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { fetchDoctors, deleteDoctor } from '../../redux/doctor/doctorSlice';

const DoctorList = () => {
  const dispatch = useDispatch();
  const doctors = useSelector((state) => state.doctor.doctors);
  const status = useSelector((state) => state.doctor.status);
  const error = useSelector((state) => state.doctor.error);

  useEffect(() => {
    dispatch(fetchDoctors());
  }, [dispatch]);

  const handleDeleteDoctor = (doctorId) => {
    dispatch(deleteDoctor(doctorId));
  };

  return (
    <div>
      {status === 'loading' && <p>Loading...</p>}
      {status === 'failed' && (
        <p>
          Error:
          {error}
        </p>
      )}

      {status === 'succeeded' && (
        <ul>
          {doctors.map((doctor) => (
            <li key={doctor.id}>
              <Link to={`/doctors/${doctor.id}`}>
                <strong>{doctor.name}</strong>
                {' '}
                -
                {' '}
                {doctor.specialization}
              </Link>
              <button type="button" onClick={() => handleDeleteDoctor(doctor.id)}>Delete</button>
            </li>
          ))}
        </ul>
      )}
    </div>
  );
};

export default DoctorList;
