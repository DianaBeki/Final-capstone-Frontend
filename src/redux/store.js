import { configureStore } from '@reduxjs/toolkit';
import doctorReducer from './doctor/doctorSlice';
import userSlice from './user/userSlice';

const store = configureStore({
  reducer: {
    doctor: doctorReducer,
    user: userSlice,
  },
});

export default store;
