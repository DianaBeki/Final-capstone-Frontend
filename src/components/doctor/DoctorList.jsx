import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { fetchDoctors } from '../../redux/doctor/doctorSlice';

const DoctorList = () => {
  const dispatch = useDispatch();
  const doctors = useSelector((state) => state.doctor.doctors);
  const status = useSelector((state) => state.doctor.status);
  const error = useSelector((state) => state.doctor.error);

  useEffect(() => {
    dispatch(fetchDoctors());
  }, [dispatch]);

  return (
    <div className="flex flex-col justify-center items-center">
      <h1 className="text-[#1F1717]">Doctors List</h1>
      <span className="text-gray-400">Choose a doctor</span>
      {status === 'loading' && <p>Loading...</p>}
      {status === 'failed' && (
        <p>
          Error:
          {error}
        </p>
      )}

      {status === 'succeeded' && (
        <ul className="gap-[2rem] md:flex md:gap-[5rem] md:mt-10">
          {doctors.map((doctor) => (
            <li key={doctor.id} className="my-[2rem] md:my-0 transition-transform transform hover:scale-110  duration-500">
              <Link to={`/doctors/${doctor.id}`} className="no-underline">
                <img src={doctor.image} alt={doctor.name} className="rounded-[50%] object-cover w-72 h-72" />
                <div className="gap-0 flex flex-col justify-center items-center md:gap-1">
                  <div className="text-[#1F1717]">
                    <strong>
                      {doctor.name}
                    </strong>
                  </div>
                  <p className="text-gray-300">................................</p>
                  <div className="text-[10px] text-gray-500 font-semi-bold md:text-[18px] md:leading-[30px]">
                    {doctor.specialization}
                  </div>
                </div>
              </Link>
            </li>
          ))}
        </ul>
      )}
    </div>
  );
};

export default DoctorList;
