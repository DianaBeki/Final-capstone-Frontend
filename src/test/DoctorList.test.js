import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { Provider } from 'react-redux';
import DoctorList from '../components/doctor/DoctorList';
import store from '../redux/store';

const render = (component) => rtlRender(
  <Provider store={store}>
    {component}
  </Provider>,
);

describe('DoctorList Component', () => {
  it('renders DoctorList component', () => {
    const { container } = render(<DoctorList />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
