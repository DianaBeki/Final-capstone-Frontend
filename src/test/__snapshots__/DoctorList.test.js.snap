// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoctorList Component renders DoctorList component 1`] = `
<div
  class="flex flex-col justify-center items-center flex-wrap"
>
  <h1
    class="text-[#1F1717]"
  >
    Doctors List
  </h1>
  <span
    class="text-gray-400"
  >
    Choose a doctor
  </span>
  <p>
    Loading...
  </p>
</div>
`;
